File: de.tototec.tools.cmvn/src/main/java/de/tototec/tools/cmvn/model/Repository.java
package de.tototec.tools.cmvn.model;

import lombok.Data;

@Data
public class Repository {
	private final String url;
	private String id;
	private boolean forPlugins = true;
	private boolean forArtifacts = true;
	private boolean forReleases = true;
	private boolean forSnapshots = false;

	public String getNormalizedRepositoryId() {
		return id != null && id.length() > 0 ? id : url.replaceAll("[\\/:\"<>|?*]", "_").replaceAll("_+", "_");
	}

}

File: de.tototec.cmvn/src/main/scala/de/tototec/cmvn/model/Repository.scala
package de.tototec.cmvn.model

import lombok.Data
//remove if not needed
import scala.collection.JavaConversions._

case class Repository(val url: String,
                      var id: String = null,
                      var forPlugins: Boolean = true,
                      var forArtifacts: Boolean = true,
                      var forReleases: Boolean = true,
                      var forSnapshots: Boolean = false) {

  def this(url: String) = this(url, id = null)

  def getNormalizedRepositoryId(): String = {
    if (id != null && id.length > 0) id else url.replaceAll("[\\/:\"<>|?*]", "_").replaceAll("_+", "_")
  }
}
